src/targets/gpu/device/include/migraphx/gpu/device/reduce.hpp

```diff
@@ -118,9 +116,11 @@ __device__ inline void dpp_reduce(float& x, sum)
                      "s_nop 1\n"
                      "v_add_f32 %0 %0 %0 row_shr:8 bank_mask:0xc\n"
                      "s_nop 1\n"
```

Comment:
What happens if we increment the n_op to 2?  Is the __syncthreads still needed?

However, overall, a much better way to implement waits for gds/lds memory transactions to become visible is:
s_waitcnt     lgkmcnt(0)

Can this allow getting rid of __syncthreads().  
Besides the __syncthreads you added in this workaround, there are also other __syncthreads in this file, which, on a superficial look, appear unnecessary. syncthreads arguably does too much, and if we can avoid it, we should...

Reply:
I have tried with increasing n_op to 2, and I still wasn't able to get the correct results.

As for the other __syncthreads(), I can try opening a separate PR if we can remove them, but for now I didn't want to modify that part of the code.